### DISCUSSION TITLE
Understanding lock profiling stack time

### DISCUSSION THREAD

**Discussion by j-bahr** [async-profiler/async-profiler#965] [Q&A] [answered]
Hi, im wondering how I can interpret the following time unit on a lock profile stack trace. 

I ran the profiler in the following configuration
```
./bin/asprof -e lock -d 10 --lock 200 -f /tmp/prof 216
```

I see output like the following 

```
--- Execution profile ---
Total samples       : 238508
skipped             : 1547 (0.65%)

--- 2554477124754 ns (35.43%), 70764 samples
  [ 0] io.airlift.stats.DecayCounter
  [ 1] io.airlift.stats.DecayCounter.add
  ...

```

How do I interpret the time value of `2554477124754 ns`? That's roughly equivalent to 42 minutes of time. My profile only lasted 10 seconds. I am running this on a machine with high thread count and highly parallel operations but what I don't know how to reason about is the time on each stack trace. Any suggestions would be great.

**Comment by apangin** [comment, marked as answer]
The value is the total sum of waiting times for all threads. It can exceed profiling duration, because it counts parallel threads waiting at the same time.
2554477124754 ns / 70764 samples = 36 ms. This means, threads were waiting 36 ms on average to acquire a lock.

Note that profiling highly contended locks with a low threshold (200 ns is definitely low) may cause additional overhead, because stack traces are collected while holding these locks. The problem is discussed in #805. We are working on the solution to this problem; meanwhile, you can simply increase threshold, e.g. `--lock 1ms`

**Reply by j-bahr** [reply]
Okay, this makes sense, thanks for the great explanation!
